wd_helper.au3

```diff
@@ -636,6 +636,199 @@ Func _WD_FrameLeave($sSession)
 	Return SetError(__WD_Error($sFuncName, $_WD_ERROR_Success), 0, $sValue)
 EndFunc   ;==>_WD_FrameLeave
 
+; #FUNCTION# ====================================================================================================================
+; Name ..........: _WD_FrameList
+; Description ...: get list of documents (window.top and all frames) from current webdriver context/tab
+; Syntax ........: _WD_FrameList($sSession[, $bReturnAsArray = True[, $sFilter = ''[, $bReturnHTML = False]]])
+; Parameters ....: $sSession            - Session ID from _WD_CreateSession
+;                  $bReturnAsArray      - By default on True function will return array, other wise string
+;                  $sFilter             - RegExp pattern to check if desired content exist in frame document HTML source, if used it returns only frames with diesired content
+;                  $bReturnHTML         - On true function will return frame document HTML source
+; Return values .: Success - array or string
+;                  Failure - "" (empty string) and sets @error to one of the following values:
+;                  - $_WD_ERROR_GeneralError
+;                  - $_WD_ERROR_Exception
+;                  - $_WD_ERROR_InvalidExpression
+;                  - $_WD_ERROR_NotFound
+; Author ........: mLipok
+; Modified ......:
+; Remarks .......: using $sFilter will set $bReturnHTML
+; Related .......: __WD_FrameList_Internal, _WD_FrameEnter, _WD_GetFrameCount
+; Link ..........:
+; Example .......: No
+; ===============================================================================================================================
+Func _WD_FrameList($sSession, $bReturnAsArray = True, $sFilter = '', $bReturnHTML = False)
+	Local Const $sFuncName = "_WD_FrameList"
+	Local Const $sParameters = 'Parameters:    ReturnAsArray=' & $bReturnAsArray & '   Filter=' & $sFilter & '   ReturnHTML=' & $bReturnHTML
+	Local $a_Result[0][6], $sStartLocation = '', $sMessage = ''
+	Local $vResult = '', $iErr = $_WD_ERROR_Success, $iExt = 0
+	Local $sStartBody_ElementID = _WD_ExecuteScript($sSession, "return window.document.body;", Default, Default, $_WD_JSON_Element)
+
+	If $sFilter <> '' Then $bReturnHTML = True
+	If @error Then
+		$iErr = $_WD_ERROR_GeneralError
```

Comment:
What if the user explicitly supplies a value of `False` for $bReturnHTML? It seems wrong to me to ignore their request. Also, I would generally expect the `@error` line to immediately follow the command that generated the error.

Reply:
I used `$bReturnHTML_Internally`

Please take a look:
[further changes](https://github.com/Danp2/au3WebDriver/pull/362/commits/3bdf5437e1f1792e68f9d69804f593ce4cd6ac52)